File: cypress/pages/EmployeeListPage.ts
import { BasePage } from './BasePage';

export class EmployeeListPage extends BasePage {
  protected url: string = '/web/index.php/directory/viewDirectory';

  searchByName(name: string): void {
    cy.get('input[placeholder="Type for hints..."]').clear().type(name);
    // Wait for autocomplete dropdown then select the first matching option
    cy.get('[role="listbox"] [role="option"]').first().click();
  }

  clickSearch(): void {
    cy.contains('button', 'Search').click();
    cy.get('.orangehrm-directory-card', { timeout: 10000 }).should('exist');
  }

  getResultRows(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.get('.orangehrm-directory-card');
  }
}

File: playwright/pages/employee_list_page.py
from playwright.sync_api import Page
from .base_page import BasePage

class EmployeeListPage(BasePage):
    URL = "/web/index.php/directory/viewDirectory"

    NAME_INPUT = 'input[placeholder="Type for hints..."]'
    SEARCH_BUTTON = 'button:has-text("Search")'
    RESULT_CARDS = '.orangehrm-directory-card'
    RESULT_COUNT = '.orangehrm-directory-result-content, .orangehrm-container'

    def visit(self):
        self.navigate(self.URL)
        self.page.wait_for_load_state("networkidle")

    def search_by_name(self, name: str):
        self.page.fill(self.NAME_INPUT, name)
        self.page.wait_for_selector('[role="listbox"] [role="option"]', timeout=5000)
        self.page.locator('[role="listbox"] [role="option"]').first.click()

    def click_search(self):
        self.page.click(self.SEARCH_BUTTON)
        # Wait for results to update — either a card appears or "No Records Found"
        self.page.wait_for_selector(
            f'{self.RESULT_CARDS}, .oxd-text:has-text("No Records")',
            timeout=15000
        )

    def get_result_rows(self):
        return self.page.locator(self.RESULT_CARDS)
